### PR TITLE
Plugin qmv

### PR DESCRIPTION
**Pull request by jlneder**
It should work as expected.

### CODE REVIEW

File: plugins/qmv [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# - Try to mimic current batch rename functionality but with correct
+#   handling of edge cases by qmv.
+#
+# Shell: POSIX compliant
+# Author: José Neder
+
+selection=${XDG_CONFIG_HOME:-$HOME/.config}/nnn/.selection
+echo $selection
+
+if [ ! -s "$selection" ] && [ -z "$1" ]; then
```
Comment (jarun) [inline plugins/qmv]: I think we can get rid of this initial check. Please consider the following:

```
if selection exists:
    ask "rename selection?"

if resp = 'y'
    rename selection
    exit

if current dir is not empty:
    qmv $PWD
```

Note: "$1" is the hovered file or dir. I think the more natural use case is visit the directory and issue batch rename.

File: plugins/qmv [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env sh
+
+# Description: Batch rename selection or current directory with qmv
+#
+# Notes:
+# - Try to mimic current batch rename functionality but with correct
```
Comment (jarun) [inline plugins/qmv]: `vidir` is probably a better candidate with numbering and all.

**Comment by jarun** [comment]
In addition to review comments, CI fails because POSIX `read` has no `-n`.

File: plugins/qmv [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+selection=${XDG_CONFIG_HOME:-$HOME/.config}/nnn/.selection
+echo $selection
+
+if [ ! -s "$selection" ] && [ -z "$1" ]; then
+   exit
+fi
+
+if [ -s "$selection" ]; then
+    printf "rename sel? "
```
[the remark this answers is not in the document]
Comment (jlneder) [reply plugins/qmv]: in nnn say rename sel, I only put the same string.

[the remark this answers is not in the document]
Comment (jarun) [reply plugins/qmv]: I guessed that. `nnn` strives to save bytes. With scripts we don't have any such restrictions.

File: plugins/qmv (hunk 1 above)
Comment (jlneder) [reply plugins/qmv]: got it

File: plugins/qmv [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    read -r -n1 resp
```
[the remark this answers is not in the document]
Comment (jlneder) [reply plugins/qmv]: I didn't think of it. Ok

**Comment by jlneder** [comment]
I push just push a version with the changes requested. I open qmv with hidden files now  but it would be better if I can somehow know if nnn is showing hidden files or not. Ideas?

**Comment by jarun** [comment]
I'll check and confirm.

Also, please consider using vidir than qmv. For newbies, vidir is much more cleaner than qmv (even the columns don't show properly in qmv and tht really put me off) .

**Comment by jlneder** [comment]
Are you talking about dual-column mode?
You should run qmv with the same tabsize that your editor:
for ex.
> qmv -otabsize=4 
FYI I'm using it in destination only mode, so, no improperly columns here.
But I'm changing the name to batchrename and generalizing the plugin to take qmv or vidir. Give me a few minutes.

**Comment by jlneder** [comment]
I just force push the changes. Tell me if it needs another change or fix.

**Comment by jarun** [comment]
If hidden files are shown in `nnn` (using `.`) we take all hidden in rename otherwise only non-hidden.

Yes, both vidir and qmv is good.

**Comment by jarun** [comment]
Please `git mv` the plugin to `renamer` and update the table.

**Comment by jlneder** [comment]
Well if you can tell me a way to know if hidden files are shown from the plugin, I would love to add it.
I just changed the name from batchrename to renamer.

**Comment by jarun** [comment]
Please add an var on top that users can set if they want to see hidden also. By default leave it without hidden.

In general, we expect users to customize plugins and use as they need. A lot of work has gone in to detect user modified plugins and give options to keep, overwrite, merge etc.

File: plugins/renamer [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    exit
+fi
+
+if [ -s "$selection" ]; then
+    printf "rename selection? "
+    read -r resp
+fi
+
+if [ "$resp" = "y" ]; then
+    # -o flag is necessary for interative editors
```
Comment (jarun) [inline plugins/renamer]: interative -> interactive

File: plugins/renamer [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    xargs -o -0 $batchrenamesel < "$selection"
+elif [ ! "$(LC_ALL=C ls -a)" = ".
```
Comment (jarun) [inline plugins/renamer]: Please drop these hacky checks and keep the code clean.

**Comment by jarun** [comment]
I'll take care of the changes. Thanks!

**Comment by jarun** [comment]
Works like a charm!

File: plugins/renamer [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+.." ]; then
```
Comment (jarun) [inline plugins/renamer]: @jlneder why is the newline necessary here?

**Pull request merged**